src/component/TraitFilter.tsx

```diff
@@ -0,0 +1,486 @@
+
+  useEffect(() => {
+
+    // const filteredArray = allTraits.map((trait) => ({ traitTitle: trait.traitTitle, searchTerm: '' }))
+    const newArray: individualChosenTraitProp[] = traitsData.map(a => ({ traitType: a.traitTitle, numberOfTraitChildrenCheck: 0, tagId: null }))
+    setIndividualChosenTrait(newArray)
+    setAllTraits(traitsData)
+  }, [traitsData])
+
+
+  // handleFilterArray
+  const handleFilterArray = (searchTerm: string, traitTitle: string) => {
+    // const []
+    if (allTraits || traitsData) {
+      const searchTermByUser = searchTerm.trim().toLowerCase()
+      const { traitProperties } = allTraits.find((prop: any) => prop.traitTitle == traitTitle)
+
+      // If user typing then perform search
+      if (searchTermByUser.length >= 1) {
+
+        const searchArrayOfTrait = traitProperties.map((word: any) => {
+          const traitTypeSearch = word.traitType.toLowerCase()
+          if (traitTypeSearch.includes(searchTermByUser)) {
+            return word;
+          }
+        }).filter((w: any) => w != undefined)
+
+
+        const newArray = allTraits.map((trait: AllTraitsProp) => {
+          if (trait.traitTitle == traitTitle) {
+            return {
+              ...trait, traitProperties: searchArrayOfTrait
+            }
+          }
+          else { return trait }
+        })
+
+
+        setAllTraits(newArray)
+        // console.log(newArray)
+
+      }
+      else {
+        // else just return original data
+        setAllTraits(traitsData)
+
+      }
+
+    }
+
+  }
+
+  const getNumberOfTraitChecked = (text: string): number => {
+    if (individualChosenTrait && individualChosenTrait.length > 0) {
+      return individualChosenTrait.find((o: any) => o.traitType == text)?.numberOfTraitChildrenCheck as number
+    }
+    return 0
+
+  }
+
+
+  // Counts every checkbox the user clicked
+
+  const [checkedTraitCount, setCheckedTraitCount] = useState<number>(0)
+
+  const [trackCheckedTraits, setTrackCheckedTraits] = useState<TrackNumberOfTraitsClicked[]>([])
+
+  // This contains every tag the user chooses
+
+
+
+  // This make sure a check box is unchecked when it's corrensponding tag was closed by a user
+  const unCheckCheckBoxIfTagIsRemoved = (prop: any, index: number) => {
+    const { traitTitle, traitType } = prop
+    const tagId = `${traitTitle}-${traitType}-${index}`
+    const isTagUnchecked = allTraitTag.some(tag => tag.tagId === tagId);
+    return isTagUnchecked
+  }
+
+  return <VStack as='section' bg='#292929'
+    position='relative'
+    w={{
+      base: '90%'
+      , md: '50%'
+      , lg: '40%'
+    }} pl='23px' pt='25px'
+    mx='auto'
+    mb='10'
+    mt='24px'
+    h='fit-content'
+    align={'flex-start'}
+    fontSize='16px'
+    spacing={'19px'}
+    sx={{
+
+      '*:focus,*:focus-visible,*:focus-within': {
+        boxShadow: '0 0 0 0px #000 !important',
+        outline: 'none'
+      },
+      '.nftcheese__allTags':
```

Comment:
instead of styling the children like this, please use chakra styling on the children components themselves